### PR TITLE
Allow unified search filtering and bump @nextcloud/vue

### PR DESCRIPTION
**Pull request by skjnldsv**
Capture d’écran_2020-08-06_08-34-04

### CODE REVIEW

File: core/src/components/UnifiedSearch/SearchFilter.vue [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	margin-right: 5px;
+	padding: 3px 8px;
+	border-radius: 1em;
+	background-color: var(--color-background-darker);
+
+	&:active,
+	&:focus,
+	&:hover {
+		background-color: var(--color-background-hover);
+	}
```
Comment (skjnldsv) [inline core/src/components/UnifiedSearch/SearchFilter.vue:74]: Here you go @jancborchardt, choose whatever you want :D

**Comment by georgehrke** [comment]
OT: Are there plans to allow search providers to support their own filters?
For events i could think of: `before: 2020`, `at: location123`

**Comment by skjnldsv** [comment]
> OT: Are there plans to allow search providers to support their own filters?
> For events i could think of: `before: 2020`, `at: location123`

You can do it already, i'm only filtering in:xxxx searches, anything else will be pass through the search.

**Review by jancborchardt — DISMISSED** [review]
As discussed, the filters are now a bit too present – more present than the results themselves.

Since it’s basically a guide for very advanced features, they can go into a 3-dot menu on the right inside the search field, which opens a menu with these filters. That way they are not in the way, people can still easily reach them via mouse, and they are also learnable

**Comment by jancborchardt** [comment]
Looks much better with the menu! :) Only: If the icon is the same for all, we don’t need the icon. Or we show the relevant app icon there. :)

**Comment by skjnldsv** [comment]
> Or we show the relevant app icon there

Would be nice, but we don't provide a unified way to get the apps icons :)
Something for 21? :grin:

**Comment by nickvergessen** [comment]
But search filters are not bound to apps... So therw is no 1-to-1 mapping to app icons

**Comment by skjnldsv** [comment]
> But search filters are not bound to apps... So therw is no 1-to-1 mapping to app icons

Exactly :)
We'd need a new entry in the api :wink:

**Comment by skjnldsv** [comment]
Let get this in, the popup fix will come from the vue library. It's irrelevant to this pr! :rocket:

**Comment by nickvergessen** [comment]
> Conflicting files
core/js/dist/install.js
core/js/dist/login.js
core/js/dist/login.js.map
core/js/dist/main.js
core/js/dist/main.js.map
core/js/dist/maintenance.js
core/js/dist/maintenance.js.map
core/js/dist/recommendedapps.js
core/js/dist/recommendedapps.js.map
core/js/dist/unified-search.js
core/js/dist/unified-search.js.map

**Comment by skjnldsv** [comment]
> /compile amend /

This is not a rebase bot!
Please wait for my rebase

**Comment by nickvergessen** [comment]
Right, i thought compile could fix it... Too long vacation 😅
Just wanted to make sure we can actually merge it tomorrow instead of having to rebase in daily traffic

**Comment by skjnldsv** [comment]
> Just wanted to make sure we can actually merge it tomorrow instead of having to rebase in daily traffic

I'm about to push alongisde a new vue release :)

**Comment by skjnldsv** [comment]
> I'm about to push alongisde a new vue release :)

DOne :)

**Comment by skjnldsv** [comment]
Needs a new nv/vue release as sidebars are buggued, see https://github.com/nextcloud/nextcloud-vue/pull/1303/

**Review by georgehrke — APPROVED** [review]
👍 Tested and works

**Comment by skjnldsv** [comment]
Let's get this in!

**Review by georgehrke — APPROVED** [review]
Still working as expected 👍

**Comment by georgehrke** [comment]
Only minor issue is the text overflow here, but let's fix that in a follow up PR:
3EC70324-DC30-46EE-8656-A944D96965D0

**Comment by skjnldsv** [comment]
> Only minor issue is the text overflow here, but let's fix that in a follow up PR:

Not sure this is a fix for server. More like a vue component fix.

**Comment by georgehrke** [comment]
Opened an issue there: https://github.com/nextcloud/nextcloud-vue/issues/1367

**Comment by skjnldsv** [comment]
Needs https://github.com/nextcloud/nextcloud-vue/pull/1368

**Comment by nickvergessen** [comment]
> ### Conflicting files
> core/src/views/UnifiedSearch.vue

:see_no_evil:

**Review by nickvergessen — APPROVED** [review]
:shark:

**Pull request merged**